automapper/path_mapper.py

```diff
@@ -0,0 +1,29 @@
+class MapPath:
+    """Represents a recursive path to an object attribute using dot notation (e.g., ob.attribute.sub_attribute)."""
+
+    def __init__(self, path: str):
+        if "." not in path:
+            raise ValueError(f"Invalid path: '{path}' does not contain '.'")
+        self.path = path
+        self.attributes = path.split(".")
+        if not len(self.attributes) >= 1:
+            raise ValueError(
+                f"Invalid path: '{path}'. Can´t reference to object attribute."
+            )
+
+        self._obj_prefix: str | None = None
```

Comment:
Is `.` really required? what if user just wants to map to source object property with different name?
E.g.
```python
fields_mapping={
     "name": MapPath("user_name")    # where user_name is just a property in source object.
}
```